### ISSUE TITLE
Non default s3 credential profiles do not work on Unified Repository Provider (kopia)

### ISSUE THREAD

**Issue by kaovilai** [vmware-tanzu/velero#6557]
**What steps did you take and what happened:**

People are reporting that non-default credentials profile do not work with kopia/unified repository provider

GetS3Credentials do not honor "profile" config field in pkg/repository/config/aws.go

Due to empty profile passed to credentials.NewSharedCredentials function
https://github.com/vmware-tanzu/velero/blob/9c8275eda756c1192d0ccdf533cdc6de0c3baaac/pkg/repository/config/aws.go#L73

inspecting go/pkg/mod of aws sdk, you will note that If empty will default to environment variable "AWS_PROFILE" or "default" if environment variable is also not set which is not desirable

```go
type SharedCredentialsProvider struct {
	// Path to the shared credentials file.
	//
	// If empty will look for "AWS_SHARED_CREDENTIALS_FILE" env variable. If the
	// env value is empty will default to current user's home directory.
	// Linux/OSX: "$HOME/.aws/credentials"
	// Windows:   "%USERPROFILE%\.aws\credentials"
	Filename string

	// AWS Profile to extract credentials from the shared credentials file. If empty
	// will default to environment variable "AWS_PROFILE" or "default" if
	// environment variable is also not set.
	Profile string

	// retrieved states if the credentials have been successfully retrieved.
	retrieved bool
}

// NewSharedCredentials returns a pointer to a new Credentials object
// wrapping the Profile file provider.
func NewSharedCredentials(filename, profile string) *Credentials {
	return NewCredentials(&SharedCredentialsProvider{
		Filename: filename,
		Profile:  profile,
	})
}
```

BSL today supports non default profiles
https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/e95ecde600af5ac5dd56e7b9ab7a2f1fc606a839/backupstoragelocation.md#L86

**What did you expect to happen:**

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 

_If you are using earlier versions:_  
Please provide the output of the following commands (Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`): 
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by Lyndon-Li** [comment]
@kaovilai Thanks for catching this issue!

**Comment by Lyndon-Li** [comment]
We also need to fix this in 1.10.x and 1.11.x
